### PR TITLE
Allowing more than two entries in affiliate credits

### PR DESCRIPTION
**Pull request by weerd**
### What does this PR do?

This PR updates the `AffiliateCredits` component to allow for more than a total of two items. Before this update, the credits would only show a single sponsor or partner, or show a sponsor and a partner, but no more than one of each total. So if there is a sponsor and 3 partners, it would only show the sponsor and the first partner in the list.

This also adds some logic to help separate each affiliate with a comma and if it is the last affiliate will prepend "and". It also ensures that if there are many affiliates which leads to a carriage return of the copy, the break doesn't happen in the middle of the affiliate's name, but breaks the whole name to the next line so it reads well.

It also does a bit of cleanup and moves the component to the utilities directory and removes the old approach of having an index file in the component directory.

### Any background context you want to provide?

Here's an example, with 4 affiliate credits.
image

But also looks fine with just one affiliate:
image

### What are the relevant tickets/cards?

Refs [Pivotal ID #161694827](https://www.pivotaltracker.com/story/show/161694827)

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: resources/assets/scss/base.scss [hunk 1]
```diff
@@ -392,3 +392,12 @@ p {
     font-weight: 400;
   }
 }
+
+// Tailwind-esk class names
+.w-1\/2 {
```
Comment (DFurnes) [inline resources/assets/scss/base.scss]: Woah. This is trippy.

File: resources/assets/components/utilities/AffiliateCredits/AffiliateCredits.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  return affiliates.length ? (
+    <div className="w-1/2 float-left">
+      In partnership with {affiliatesString}
+    </div>
+  ) : null;
+};
+
+AffiliateCredits.propTypes = {
+  affiliateSponsors: PropTypes.arrayOf(PropTypes.object).isRequired,
```
Comment (DFurnes) [inline resources/assets/components/utilities/AffiliateCredits/AffiliateCredits.js]: Do we want these to be optional and default to `[]` if not provided?

File: resources/assets/scss/base.scss (hunk 1 above)
Comment (weerd) [reply resources/assets/scss/base.scss]: Yeah! Something I saw tailwind doing, but makes it nice to use fractions in classnames 💃

File: resources/assets/components/utilities/AffiliateCredits/AffiliateCredits.js (hunk 2 above)
Comment (weerd) [reply resources/assets/components/utilities/AffiliateCredits/AffiliateCredits.js]: oh yeah, I dig that. I'll make that change and then mergy merge. For both `affiliateSponsors` and `affiliatePartners` right?

Comment (DFurnes) [reply resources/assets/components/utilities/AffiliateCredits/AffiliateCredits.js]: Yeah, that makes sense to me! 👍

**Pull request merged**